arch/arm/core/aarch64/macro_priv.inc

```diff
@@ -9,31 +9,64 @@
 
 #ifdef _ASMLANGUAGE
 
-.macro z_arm64_get_cpu_id xreg0
+GDATA(_curr_cpu)
+GDATA(_kernel)
+
+/*
+ * Get CPU id
+ */
+
+.macro get_cpu_id xreg0
 	mrs	\xreg0, mpidr_el1
 	/* FIMXME: aff3 not taken into consideration */
 	ubfx	\xreg0, \xreg0, #0, #24
 .endm
+
+/*
+ * Get CPU pointer
+ */
+
+.macro get_cpu xreg0, xreg1
+	ldr	\xreg0, =_curr_cpu
+	get_cpu_id \xreg1
+	add	\xreg0, \xreg0, \xreg1, lsl #3
```

Comment:
It was just easier to me to rely on `_curr_cpu`. But considering that `_curr_cpu` is only used here, I'll push later another PR to remove it entirely but for now I'd like to keep this PR as is otherwise we have to rebase #32854 that is using this code.